File: client-scalaz/src/main/scala/org/openapitools/client/api/PermissionResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import PermissionResponse._

case class PermissionResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
read: Option[Boolean],
write: Option[Boolean],
delete: Option[Boolean],
add: Option[Boolean],
returning: Option[String])

object PermissionResponse {
  import DateTimeCodecs._

  implicit val PermissionResponseCodecJson: CodecJson[PermissionResponse] = CodecJson.derive[PermissionResponse]
  implicit val PermissionResponseDecoder: EntityDecoder[PermissionResponse] = jsonOf[PermissionResponse]
  implicit val PermissionResponseEncoder: EntityEncoder[PermissionResponse] = jsonEncoderOf[PermissionResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/permission_response.clj
(ns sirqul-io-t-platform.specs.permission-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def permission-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :read) boolean?
   (ds/opt :write) boolean?
   (ds/opt :delete) boolean?
   (ds/opt :add) boolean?
   (ds/opt :returning) string?
   })

(def permission-response-spec
  (ds/spec
    {:name ::permission-response
     :spec permission-response-data}))
